lib/moonshot/command.rb

```diff
@@ -23,6 +23,10 @@ def parser
           Moonshot.config.interactive_logger = InteractiveLogger.new(debug: true) if v
         end
 
+        o.on('-s', '--skip-ci-status', 'Skip checks on CI jobs') do
```

Comment:
We added the ability for plugins and mechanisms to hook into the CLI parser when dispatching a command, we can use that here to avoid having this option exposed to classes that aren't going to use it.

That might look something like (assuming this is used during deploy):

In github_release.rb:
```ruby
def initialize(skip_ci_status: false)
  @skip_ci_status = skip_ci_status
end
```

```ruby
def deploy_cli_hook(parser)
  parser.on('-s', '--skip-ci-status', 'Skips checks on CI jobs', FalseClass) do |value|
    @skip_ci_status = value
  end
end
```

In this way, you completely own the configuration of the mechanism within itself.

Reply:
I was start this feature with this approach, actually I like it to mush because I lean toward keeping every module as a standalone and ready for plug and play.

Then I moved to current approach based on 2 reasons:

1.    I would like to see that flag on --help command e.g.
```
moonshot build --help
Usage: moonshot build VERSION
    -v, --[no-]verbose               Show debug logging
    -s, --skip-ci-status             Skip checks on CI jobs
    -n, --environment=NAME           Which environment to operate on.
        --[no-]interactive-logger    Enable or disable fancy logging
```
2.    I thought all plugins,  added by Acquia team are part of command.rb as a backbone of Moonshot.
So now what you prefer?

Reply:
Perhaps moonshot could be enhanced to call the proper hooks when command help is requested.

Reply:
I agree with @pdrakeweb, but we can do that as a follow-up. We have other flags that are missing for core features because of the same reason, like CodeDeploy.

Reply:
Could you take a look again @askreet @pdrakeweb also I think we have to add a story in JIra or a note to be done later for enhancing moonshot command line when --help flag is requested.

Reply:
No need for --help hock .. I think now its working once plugin registered .. PR now updated with steps for checking --help flag and we can see
```
moonshot deploy --help 
Usage: moonshot deploy VERSION
    -v, --[no-]verbose               Show debug logging
    -n, --environment=NAME           Which environment to operate on.
        --[no-]interactive-logger    Enable or disable fancy logging
        --[no-]interactive           Use interactive prompts.
    -s, --skip-ci-status             Skips checks on CI jobs
        --ignore-app-stop-failures   Continue deployment on ApplicationStop failures
```

Reply:
@askreet Could you take a look again,, also PR description updated